polaris-react/src/components/Modal/components/CloseButton/CloseButton.scss

```diff
@@ -31,4 +31,13 @@
   &.titleHidden {
     margin: var(--p-space-2);
   }
+
+  #{$se23} & {
+    margin-top: calc(-1 * var(--p-space-1));
+    margin-bottom: calc(-1 * var(--p-space-1));
```

Comment:
I think this was done to ensure we still have a large clickable area on the close button. Looks good to me.